plugins/search/src/components/SearchResult/SearchResult.tsx

```diff
@@ -0,0 +1,230 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useState, useEffect } from 'react';
+import { useAsync } from 'react-use';
+
+import { makeStyles, Typography, Grid, Divider } from '@material-ui/core';
+import { Table, TableColumn } from '@backstage/core';
+import { FilterButton, Filters } from '../Filters';
+
+import SearchApi from '../../apis';
+
+const useStyles = makeStyles(() => ({
+  searchTerm: {
+    background: '#eee',
+    padding: '0.2%',
+    borderRadius: '10%',
+  },
+  tableHeader: {
+    marginTop: '10px',
+    display: 'flex',
+  },
+  divider: {
+    width: '1px',
+    margin: '0 20px',
+    padding: '15px 0',
+  },
+}));
+
+type SearchResultProps = {
+  currentTarget?: string;
+};
+
+type TableHeaderProps = {
+  searchTerm?: string;
+  numberOfSelectedFilters: number;
+  numberOfResults: number;
+  handleToggleFilters: () => any;
+};
+
+type Filters = {
+  selected: string;
+  checked: Array<string | null>;
+};
+
+// TODO: move out column to make the search result component more generic
+const columns: TableColumn[] = [
+  {
+    title: 'Component Id',
+    field: 'name',
+    highlight: true,
+  },
+  {
+    title: 'Description',
+    field: 'description',
+  },
+  {
+    title: 'Owner',
+    field: 'owner',
+  },
+  {
+    title: 'Kind',
+    field: 'kind',
+  },
+  {
+    title: 'LifeCycle',
+    field: 'lifecycle',
+  },
+];
+
+const TableHeader = ({
+  searchTerm,
+  numberOfSelectedFilters,
+  numberOfResults,
+  handleToggleFilters,
+}: TableHeaderProps) => {
+  const classes = useStyles();
+
+  return (
+    <div className={classes.tableHeader}>
+      <FilterButton
+        numberOfSelectedFilters={numberOfSelectedFilters}
+        handleToggleFilters={handleToggleFilters}
+      />
+      <Divider className={classes.divider} orientation="vertical" />
+      <Grid item sm={12}>
+        {searchTerm ? (
+          <Typography variant="h6">
+            {`${numberOfResults} `}
+            {numberOfResults > 1 ? `results for ` : `result for `}
+            <span className={classes.searchTerm}>"{searchTerm}"</span>{' '}
+          </Typography>
+        ) : (
+          <Typography variant="h6">{`${numberOfResults} results`}</Typography>
+        )}
+      </Grid>
+    </div>
+  );
+};
+
+const SearchResult = ({ currentTarget }: SearchResultProps) => {
+  const [showFilters, toggleFilters] = useState(false);
+  const [filters, setFilters] = useState<Filters>({
+    selected: 'All',
+    checked: [],
+  });
+
+  const [filteredResults, setFilteredResults] = useState<Array<object>>([]);
+
+  const searchApi = new SearchApi();
+
+  const { loading, error, value: results } = useAsync(() => {
+    return searchApi.getSearchData();
```

Comment:
this should really return something that has a typescript type, so we can avoid the use of the risky `any` below, and surprising `object` above